compiler/ballerina-lang/src/main/java/io/ballerina/projects/Document.java

```diff
@@ -53,4 +57,48 @@ public SyntaxTree syntaxTree() {
     public TextDocument textDocument() {
         return this.documentContext.textDocument();
     }
+
+    /** Returns an instance of the Document.Modifier.
+     *
+     * @return  module modifier
+     */
+    public Modifier modify() {
+        return new Modifier(this);
+    }
+
+    /**
+     * Inner class that handles Document modifications.
+     */
+    public static class Modifier {
+        private String content;
```

Comment:
This should hold all the attributes from the old document and apply() should blindly create the new document from these attributes.